martypy/WebSocket.py

```diff
@@ -67,7 +77,36 @@ def _sendUpgradeReq(self) -> None:
 
     def service(self) -> None:
         # Get any data
-        rxData = self.sock.recv(self.maxSocketBytes)
+        reconnectRequired = False
+        try:
+            rxData = self.sock.recv(self.maxSocketBytes)
+        except OSError as excp:
+            logger.debug(f"WebSocket OSError {excp}")
+            if not self.autoReconnect:
+                raise excp
+            reconnectRequired = True
+        except Exception as excp:
+            logger.debug(f"WebSocket exception {excp}")
+            if not self.autoReconnect:
+                raise excp
+            reconnectRequired = True
```

Comment:
Are we always expecting an `OSError` and is the other `except` clause merely for debugging? Or why do we have both clauses if they do the same thing? Would it be bad to merge them and avoid duplication?
```suggestion
        except Exception as excp:
            logger.debug(f"WebSocket exception {excp}")
            if not self.autoReconnect:
                raise excp
            reconnectRequired = True
```

Reply:
I'm happy to merge them - I was originally thinking of handling the more general exception differently but couldn't think of a better thing to do.